Automattic/jetpack
# Slideshow: Make sure autoplay works on the frontend

Fixes https://github.com/Automattic/jetpack/issues/36973

## Proposed changes:

* This PR ensures that Slideshow autoplay works on the frontend for sites who have this enabled.  
* Specifically, within the `Save` function a newly added div wrapping the slideshow has been removed, and the relevant blockProps (which only includes the className), added back in directly. Several changes were made to `slideshow.js` and `edit.js` to add the className back in as well (everywhere it was before except the MediaPlaceholder where it wasn't necessary - this is the placeholder when adding the block and no images are added yet).

Note - an additional issue was reported in the linked issue (images following the movement of the mouse when autoplay enabled), which has now been moved to a new issue here: https://github.com/Automattic/jetpack/issues/37407

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/36973

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* To replicate the issue, on trunk or using a test site with the most recent live Jetpack version, add a slideshow block to a post. Enable autoplay from the settings sidebar, and then save the post. Preview or publish and view the post on the front-end, and you'll notice that autoplay is not working.
* To test the fix, apply this PR and follow the same steps as above. Autoplay should now work.
* This should also be the case on Atomic and Simple sites using the testing instructions in the generated comment below.
* To test that the block deprecation version addition works, after applying the PR open up the editor with one of the previously added Slideshow blocks, and it should continue to work with no 'This block contains unexpected or invalid content' messages. Open up the published or preview page for that post and the slideshow's autoplay should now be working.
* Note - you will need to 'save' the post first in the editor, which may require making a change -- this is expected behaviour (I found toggling the autoplay settings to a different duration and back again was enough to allow me to re-save the post). This will then allow the front-end to correctly show the fixed slideshow autoplaying.
